File: src/sample08/contas/AuditoriaInterna.java
package sample08.contas;

import java.util.ArrayList;
import java.util.List;

public class AuditoriaInterna {
    private final List<Tributavel> tributaveis = new ArrayList<>();

    public void adicionar(Tributavel t) {
        tributaveis.add(t);
    }

    public double calcularTributos() {
        return tributaveis
                .stream()
                .mapToDouble(Tributavel::calcularTributos)
                .sum();
    }
}

File: src/sample11/AuditoriaInterna.ts
import { Tributavel } from "./Tributavel";

export class AuditoriaInterna {
  tributaveis: Tributavel[];

  constructor(tributaveis: Tributavel[]) {
    this.tributaveis = tributaveis;
  }

  adicionar(t: Tributavel): void {
    this.tributaveis.push(t);
  }

  calcularTributos(): number {
    return this.tributaveis
      .map((t) => t.calcularTributo())
      .reduce((sum, curr) => sum + curr);
  }
}
